### PR TITLE
Added automatic installation of GDB to build.sh

### PR DESCRIPTION
**Pull request by marvhus**
Added commands for yay, pamac, pacman, apt, apt-get, zypper, dnf, apk, emerge, and nix-env.

Moved long printf messages to more readable variables.

Added some printing during build.

### CODE REVIEW

**Comment by marvhus** [comment]
> So we expect an average contributor to gf to have no idea how to use their own Linux distro?

A few weeks ago I suggested gf to someone, but then they just immediately gave up on installing it because the build script failed, and it likely was just GDB missing.  Also, not everyone who will use this build script will be contributing to the project since there is no pre-built binaries, etc, so they have to build it themselves.  I got tired of them just complaining about it, so I just added it.  
TL;DR: People are lazy, and I got tired of hearing them complain.
